base/log_redactor_userdata.go

```diff
@@ -0,0 +1,37 @@
+package base
+
+import "fmt"
+
+const (
+	userDataPrefix = "<ud>"
+	userDataSuffix = "</ud>"
+)
+
+// RedactUserData is a global toggle for UserData redaction.
+var RedactUserData = false
+
+// UserData implements the Redactor interface for logging purposes.
+type UserData string
+
+// Compile-time interface check.
+var _ Redactor = UserData("")
```

Comment:
Here I'm forcing the compiler to check `UserData` satisfies `Redactor`. This is more useful in scenarios where the interface and implementation are in separate packages, but still a useful thing to have around IMO.

https://medium.com/@matryer/golang-tip-compile-time-checks-to-ensure-your-type-satisfies-an-interface-c167afed3aae